app/src/main/java/com/babylon/wallet/android/domain/model/TransactionHistoryItem.kt

```diff
@@ -0,0 +1,215 @@
+            when (it) {
+                is BalanceChange.FungibleBalanceChange -> it.balanceChange.signum() == -1
+                is BalanceChange.NonFungibleBalanceChange -> it.removedIds.isNotEmpty()
+            }
+        }
+
+    val unknownTransaction: Boolean
+        get() = transactionClass == null
+
+    val noBalanceChanges: Boolean
+        get() = deposited.isEmpty() && withdrawn.isEmpty()
+}
+
+data class HistoryFilters(
+    val start: ZonedDateTime? = null,
+    val transactionType: TransactionType? = null,
+    val resource: Resource? = null,
+    val transactionClass: TransactionClass? = null,
+    val sortOrder: SortOrder = SortOrder.Desc
+) {
+
+    enum class SortOrder {
+        Asc, Desc
+    }
+
+    enum class TransactionType {
+        DEPOSIT, WITHDRAWAL
+    }
+
+    val isAnyFilterSet: Boolean
+        get() = transactionType != null || resource != null || transactionClass != null
+}
+
+sealed interface BalanceChange {
+
+    val asset: Asset?
+    val entityAddress: String
+
+    data class FungibleBalanceChange(
+        val balanceChange: BigDecimal,
+        override val entityAddress: String,
+        override val asset: Asset.Fungible? = null
+    ) : BalanceChange
+
+    data class NonFungibleBalanceChange(
+        val removedIds: List<String>,
+        val addedIds: List<String>,
+        override val entityAddress: String,
+        override val asset: Asset.NonFungible? = null
+    ) : BalanceChange
+}
+
+fun ManifestClass.toTransactionClass(): TransactionClass {
+    return when (this) {
+        ManifestClass.general -> TransactionClass.General
+        ManifestClass.transfer -> TransactionClass.Transfer
+        ManifestClass.poolContribution -> TransactionClass.PoolContribution
+        ManifestClass.poolRedemption -> TransactionClass.PoolRedemption
+        ManifestClass.validatorStake -> TransactionClass.ValidatorStake
+        ManifestClass.validatorUnstake -> TransactionClass.ValidatorUnstake
+        ManifestClass.validatorClaim -> TransactionClass.ValidatorClaim
+        ManifestClass.accountDepositSettingsUpdate -> TransactionClass.AccountDespositSettingsUpdate
+    }
+}
+
+fun TransactionClass.toManifestClass(): ManifestClass {
+    return when (this) {
+        TransactionClass.General -> ManifestClass.general
+        TransactionClass.Transfer -> ManifestClass.transfer
+        TransactionClass.PoolContribution -> ManifestClass.poolContribution
+        TransactionClass.PoolRedemption -> ManifestClass.poolRedemption
+        TransactionClass.ValidatorStake -> ManifestClass.validatorStake
+        TransactionClass.ValidatorUnstake -> ManifestClass.validatorUnstake
+        TransactionClass.ValidatorClaim -> ManifestClass.validatorClaim
+        TransactionClass.AccountDespositSettingsUpdate -> ManifestClass.accountDepositSettingsUpdate
+    }
+}
+
+fun TransactionBalanceChanges.toDomainModel(assets: List<Asset>): List<BalanceChange> {
```

Comment:
minor: not hard requirement - these mappings `toDomainModel` and the `TransactionClass` enum class better in a separate file for mappings under the `data/repository/stream/`?